### PR TITLE
Supporting macosx older than 10.15

### PR DESCRIPTION
**Pull request by freddyaboulton**
#### Reference Issues/PRs

Fixes #723 

#### What does this implement/fix? Explain your changes.

Builds wheels for mac osx 10.14 and 10.15 in the deploy-to-pypi github action.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Review by mloning — COMMENTED** [review]
Thanks @freddyaboulton for the PR! Seems to work nicely! 

From what I understand, we need to change both the [build-and-test workflow](https://github.com/alan-turing-institute/sktime/blob/fd927f3c8109e2dd033f6747fc6a9c229f36f42e/.github/workflows/build-and-test.yml) and the upload-to-pypi workflow.

**Comment by freddyaboulton** [comment]
@mloning I updated the build-and-test workflow! Thanks for the tip!

**Review by mloning — DISMISSED** [review]
Thanks @freddyaboulton - looks all good to me now. 

Could you share a link where you found out about using `MACOS_DEPLOYMENT_TARGET`? Quickly searched for it but couldn't find any official documentation on it.

**Comment by mloning** [comment]
Not sure about the build failures now, seem unrelated to this PR. Will look into this.

**Comment by freddyaboulton** [comment]
@mloning Thanks for the approval and hunting down the pystan issue! 

I learned about the `MACOSX_DEPLOYMENT_TARGET` env variable from this [thread](https://github.com/actions/virtual-environments/issues/1256#issuecomment-770270252)!

**Comment by fkiraly** [comment]
Fascinating ... https://en.wikipedia.org/wiki/Linus%27s_law

**Comment by riddell-stan** [comment]
PyStan 3.0 just appeared which makes backwards incompatible changes, https://discourse.mc-stan.org/t/pystan-3-0-is-released/21500

You likely want to pin your version of PyStan to a version less than 3.

**Comment by mloning** [comment]
Thanks @riddell-stan!

Pinned the version here: #751

**Comment by freddyaboulton** [comment]
Thank you @mloning !

**Comment by hoesler4dh** [comment]
Hey @freddyaboulton,
thanks for helping with this issue. I did a litte bit of experiments on my own and it seems like simply setting `MACOSX_DEPLOYMENT_TARGET` is not enough.
You actually have to compile python with this env var set. Here is an old blog post on this: https://lepture.com/en/2014/python-on-a-hard-wheel

I can confirm this works.
After installing python using pyenv on my mac 10.14 with `MACOSX_DEPLOYMENT_TARGET=10.13`, bdist_wheel was producing a 10.13 wheel.

For the workflow it might be sufficient to just install an official mac pkg, which seems to be compiled against 10.9:
https://www.python.org/ftp/python/3.8.8/python-3.8.8-macosx10.9.pkg

So I guess something like this could work:

```
    strategy:
      matrix:
        os: [ macos-latest ]
        python-version: [ 3.6.9, 3.7.9, 3.8.8 ]
    steps:
      - uses: actions/checkout@v2
      - name: Download Python
        run: |
          curl https://www.python.org/ftp/python/${{ matrix.python-version }}/python-${{ matrix.python-version }}-macosx10.9.pkg -o "python.pkg"
      - name: Install Python
        run: |
          sudo installer -pkg python.pkg -target /
```

**Comment by hoesler4dh** [comment]
fixes #724

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -53,22 +53,22 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest]
-        python-version: [3.6, 3.7, 3.8]
+        python-version: [ 3.6.8, 3.7.9, 3.8.8 ]
```
Comment (mloning) [inline .github/workflows/build-and-test.yml:56]: Do we have to pin the exact Python version here? I'm assuming it's necessary for the URL in the curl command below?

**Review by mloning — COMMENTED** [review]
Thanks for figuring this out!

Why is macOS 10.9 sufficient here if we want 10.14?

I'll try to get this in before the next release.

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (hoesler4dh) [reply .github/workflows/build-and-test.yml:56]: If I’m not mistaken, the 10.9 python distribution is the official release for all macos versions >= 10.9. So I was thinking, that we just could use that. If you want to limit to 10.14, we need to compile Python for the target.

Regarding the python version for the curl command, I don’t think omitting the patch version would work.

**Comment by fkiraly** [comment]
@freddyaboulton , @hoesler4dh, this PR is still open - is this superseded, or is this still work in progress?
Or, would it be ready for review/merge?

**Comment by mloning** [comment]
Closed for now due to inactivity. I'm happy to revisit the idea if this is still relevant.

**Pull request closed without merging**